pyvista/core/filters/poly_data.py

```diff
@@ -2144,7 +2144,7 @@ def multi_ray_trace(
         )
         if retry:
             # gather intersecting rays in lists
-            loc_lst, ray_lst, tri_lst = [arr.tolist() for arr in [locations, index_ray, index_tri]]
+            loc_lst, ray_lst, tri_lst = (arr.tolist() for arr in [locations, index_ray, index_tri])
```

Comment:
I'm a bit meh on this one. Sure, discarding the list is semantically worse. But it's the same amount of characters and even slightly faster than using the generator expression (214 ns vs 277 ns on my system for a dummy test, so hardly critical either way).

Reply:
I would split this up into three lines/statements. Comprehensions are great, but they don't need to be used everywhere as they can reduce the readability of code... like here IMO